cluster/src/dunit/scala/io/snappydata/cluster/QueryRoutingDUnitTest.scala

```diff
@@ -1284,4 +1283,56 @@ class QueryRoutingDUnitTest(val s: String)
       i4 = i4 + 1
     }
   }
+
+  def testColumnAttributes: Unit = {
+    val netPort1 = AvailablePortHelper.getRandomAvailableTCPPort
+    vm2.invoke(classOf[ClusterManagerTestBase], "startNetServer", netPort1)
+    val conn = getANetConnection(netPort1)
+    val stmt = conn.createStatement()
+
+    stmt.executeUpdate("create database db1")
+    stmt.executeUpdate("create database db2")
+
+    stmt.executeUpdate("create table db1.t1 (c1 integer, c2 integer)")
+    stmt.executeUpdate("create table db2.t1 (c1 integer, c2 integer)")
+
+    val rs = stmt.executeQuery("select * from db1.t1")
+    assert(rs.getMetaData.getSchemaName(1).equalsIgnoreCase("db1"),
+      s"expected db1 but received ${rs.getMetaData.getSchemaName(1)}")
+    assert(rs.getMetaData.getTableName(1).equalsIgnoreCase("t1"),
+      s"expected t1 but received ${rs.getMetaData.getTableName(1)}")
+    assert(rs.getMetaData.getColumnCount.equals(2))
+    rs.close()
+
+    val rs2 = stmt.executeQuery(
+      "select a.c1, b.c2, 'static value' as c3 from db1.t1 a join db2.t1 b on a.c1 = b.c1")
+    assert(rs2.getMetaData.getSchemaName(1).equalsIgnoreCase("db1"),
+      s"expected db1 but received ${rs2.getMetaData.getSchemaName(1)}")
+    assert(rs2.getMetaData.getTableName(1).equalsIgnoreCase("t1"),
+      s"expected t1 but received ${rs2.getMetaData.getSchemaName(2)}")
+    assert(rs2.getMetaData.getSchemaName(2).equalsIgnoreCase("db2"),
+      s"expected db2 but received ${rs2.getMetaData.getSchemaName(2)}")
+    assert(rs2.getMetaData.getTableName(2).equalsIgnoreCase("t1"),
+      s"expected t1 but received ${rs2.getMetaData.getTableName(2)}")
+    assert(rs2.getMetaData.getSchemaName(3).equalsIgnoreCase(""),
+      s"expected blank but received ${rs2.getMetaData.getSchemaName(3)}")
+    assert(rs2.getMetaData.getTableName(3).equalsIgnoreCase(""),
+      s"expected blank but received ${rs2.getMetaData.getTableName(3)}")
+    assert(rs2.getMetaData.getColumnCount.equals(3),
+      s"expected column count to be 3 but received ${rs2.getMetaData.getColumnCount}")
+    rs2.close()
+
+    val rs3 = stmt.executeQuery("select c1, count(c2) from db1.t1 group by c1")
+    assert(rs3.getMetaData.getSchemaName(1).equalsIgnoreCase("db1"),
+      s"expected db1 but received ${rs3.getMetaData.getSchemaName(1)}")
+    assert(rs3.getMetaData.getTableName(1).equalsIgnoreCase("t1"),
+      s"expected t1 but received ${rs3.getMetaData.getTableName(1)}")
+    assert(rs3.getMetaData.getSchemaName(2).equalsIgnoreCase(""),
+      s"expected blank but received ${rs3.getMetaData.getSchemaName(2)}")
+    assert(rs3.getMetaData.getTableName(2).equalsIgnoreCase(""),
+      s"expected blank but received ${rs3.getMetaData.getTableName(2)}")
+    assert(rs3.getMetaData.getColumnCount.equals(2),
+      s"expected column count to be 2 but received ${rs3.getMetaData.getColumnCount}")
+    rs3.close()
+  }
```

Comment:
As discussed add a test that covers views.  Also add a test for PreparedStatement as well.